src/ArrayObject/LegacyImplementation.php

```diff
@@ -33,7 +40,7 @@
  *
  * Extends version-specific "abstract" implementation.
  */
-class ArrayObject implements IteratorAggregate, ArrayAccess, Serializable, Countable
+class LegacyImplementation implements IteratorAggregate, ArrayAccess, Serializable, Countable
```

Comment:
Could we mark this as `@deprecated` perhaps? People should not reference it directly, IMO

Reply:
This, and your request to mark the PHP81Implementation classes as `@internal`, is a bit... difficult.

Essentially, these become the actual public API - I'm aliasing the original class name to the appropriate implementation.

If we mark one as "deprecated", then the target class shows up as deprecated when using that version. If we show one as "internal", then the target class shows up as internal for that PHP version.

I think it might need to be a class-level docblock _note_ on each, indicating what each is, and that users should extend the target class instead.

Thoughts?

Reply:
> Essentially, these become the actual public API - I'm aliasing the original class name to the appropriate implementation.

This is a problem if anyone references the `LegacyImplementation` instead of `ArrayObject`. This **will** happen.

Reply:
It's a problem for anybody extending the aliased class (in this case, `Laminas\Stdlib\ArrayObject`.

With your proposal:

- Users under PHP versions prior to 8.1 will get deprecation notice due to the `@deprecated` annotation at the class level.
- Users under PHP version 8.1+ will get a notice for using a class marked internal, due to the `@internal` annotation at the class level.

Basically, these _are not_ internal or deprecated - they are the actual public implementations.